### PR TITLE
Added favoriting project from project details view functionality

### LINKED ISSUE #56: Mark as favorite in project details
## Expected

Clicking the start :star: in the project details page marks/unmarks the project as favorite

Screenshot from 2021-08-27 08-59-01

## Actual

The star is only a visual indicator, and not an actual button.

Screenshot from 2021-08-27 09-00-10

### PR DESCRIPTION
**Pull request by Alexamakans**
- \[x] I've added a new note in the `CHANGELOG.md` file, according to docs:
  https://iver-wharf.github.io/#/development/changelogs/writing-changelogs

## Summary

Added functionality to click the star in the project details view to favorite/unfavorite a project. 
image

## Motivation

Not being able to favorite a project while looking at the details for it was weird. Now we can.

Closes #56

### CODE REVIEW

File: src/app/projects/project-details/project-details.component.html [hunk 1]
```diff
@@ -4,9 +4,8 @@
       <div class="top">
         <span class="breadcrumbs"><a routerLink="/">Projects &gt;</a>
           <h4>{{project.name}}</h4>
-          <i [ngClass]="(localStorageProjectsService.projectIsFavorite(project.projectId))
-            ? 'pi pi-star favorite-project-icon' : 'pi pi-star-o favorite-project-icon'">
-          </i>
+          <i class="icon" (click)="onFavoriteIconClicked($event)" [ngClass]="(localStorageProjectsService.projectIsFavorite(project.projectId))
+            ? 'pi pi-star favorite-project-icon' : 'pi pi-star-o favorite-project-icon'"></i>
```
Comment (jilleJr) [inline src/app/projects/project-details/project-details.component.html:8]: Clickable items should be `<a>`, `<button>`, `<input>`, or have `role="button"`.

Suggest changing to `<button>`

**Review by jilleJr — CHANGES_REQUESTED** [review]

File: src/app/projects/project-details/project-details.component.html (hunk 1 above)
Comment (Alexamakans) [reply src/app/projects/project-details/project-details.component.html]: Changed to `<button>` with  `<i>` inside.
Not sure if the way I solved the styling is good or not.

**Review by fredx30 — APPROVED** [review]
Good stuff.

**Review by jilleJr — CHANGES_REQUESTED** [review]

**Review by jilleJr — APPROVED** [review]
LGTM!

**Pull request merged**